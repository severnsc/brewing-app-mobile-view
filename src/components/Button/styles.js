import { StyleSheet } from "react-native";
import { lightGray, gray, primary, success, danger } from "../../constants";

export default StyleSheet.create({
  button: {
    borderColor: lightGray,
    borderWidth: 1,
    alignItems: "center",
    padding: 10,
    backgroundColor: gray
  },
  primary: {
    backgroundColor: primary
  },
  secondary: {
    backgroundColor: "white"
  },
  success: {
    backgroundColor: success
  },
  danger: {
    backgroundColor: danger
  },
  round: {
    borderRadius: 100
  },
  circle: {
    width: 50,
    height: 50,
    borderRadius: 100,
    justifyContent: "center"
  },
  hideBorder: {
    borderWidth: 0
  }
});
