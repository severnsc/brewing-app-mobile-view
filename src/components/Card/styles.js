import { StyleSheet } from "react-native";

export default StyleSheet.create({
	card: {
		padding: 5,
		height: 80,
		justifyContent: "space-between",
		shadowColor: "#000",
		shadowOffset: { width: 4, height: 6 },
		shadowOpacity: 0.2,
		shadowRadius: 2,
		backgroundColor: "white"
	},
	row: {
		flexDirection: "row",
		justifyContent: "space-between"
	}
});
