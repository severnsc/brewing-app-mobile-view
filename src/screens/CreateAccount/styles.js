import { StyleSheet } from "react-native";
import { white } from "../../constants";

export default StyleSheet.create({
	container: {
		justifyContent: "center",
		alignItems: "center",
		backgroundColor: white,
		margin: 5,
		top: 100
	},
	form: {
		width: "100%",
		paddingHorizontal: 20,
		paddingVertical: 10
	},
	input: {
		marginVertical: 10
	}
});
