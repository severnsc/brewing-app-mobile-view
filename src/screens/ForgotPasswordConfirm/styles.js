import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5,
    marginTop: 100
  },
  text: { fontSize: 24, textAlign: "center" }
});
