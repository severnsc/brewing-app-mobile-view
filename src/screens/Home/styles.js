import { StyleSheet } from "react-native";

export default StyleSheet.create({
	container: {
		flex: 1,
		justifyContent: "space-around",
		alignItems: "center",
		padding: 5
	},
	buttonRow: {
		width: "100%"
	},
	button: {
		marginTop: 10
	}
});
