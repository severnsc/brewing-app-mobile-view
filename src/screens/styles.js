import { StyleSheet } from "react-native";
import { white } from "../constants";

export default StyleSheet.create({
  container: {
    alignItems: "center",
    justifyContent: "center",
    top: 100,
    margin: 5,
    backgroundColor: white
  },
  form: {
    width: "100%",
    paddingHorizontal: 20,
    paddingVertical: 10
  },
  input: {
    marginVertical: 10
  }
});
