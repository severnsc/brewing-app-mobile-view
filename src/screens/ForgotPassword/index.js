import React from "react";
import PropTypes from "prop-types";
import { GradientView, Form, TextInput, Button, Text } from "../../components";
import {
  KeyboardAvoidingView,
  ActivityIndicator,
  AlertIOS
} from "react-native";
import styles from "../styles";
import { NETWORK_ERROR, INVALID_EMAIL } from "../../constants";

const ForgotPassword = ({ onSubmit, isError }) => (
  <GradientView>
    <KeyboardAvoidingView style={styles.container}>
      <Form
        onSubmit={onSubmit}
        style={styles.form}
        initialValues={[{ id: "1", value: "" }, { id: "2", value: false }]}
      >
        {(values, _onChange, _onSubmit) => {
          const value = values[0].value;
          const loading = values[1].value;
          const onChange = value => _onChange("1", value);
          const onSubmit = ({ value }) => {
            _onChange("2", true);
            _onSubmit(value)
              .then(() => _onChange("2", false))
              .catch(() => {
                _onChange("2", false);
                AlertIOS.alert("Error!", NETWORK_ERROR);
              });
          };
          return (
            <React.Fragment>
              <TextInput
                errorText={isError && INVALID_EMAIL}
                onChange={onChange}
                label="Email"
                value={value}
              />
              {loading ? (
                <ActivityIndicator />
              ) : (
                <Button value="Send Email" onPress={onSubmit} />
              )}
            </React.Fragment>
          );
        }}
      </Form>
    </KeyboardAvoidingView>
  </GradientView>
);

ForgotPassword.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  isError: PropTypes.bool
};

ForgotPassword.defaultProps = {
  onSubmit: () => {},
  isError: false
};

export default ForgotPassword;
