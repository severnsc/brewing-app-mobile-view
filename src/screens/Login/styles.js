import { StyleSheet } from "react-native";
import { white } from "../../constants";
import styles from "../styles";

export default StyleSheet.create({
  ...styles,
  forgotPassword: {
    backgroundColor: white,
    borderColor: white
  }
});
