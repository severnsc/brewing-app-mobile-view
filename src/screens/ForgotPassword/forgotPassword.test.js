import React from "react";
import ForgotPassword from ".";
import { shallow } from "enzyme";
import styles from "../styles";
import { AlertIOS } from "react-native";
import { NETWORK_ERROR, INVALID_EMAIL, white } from "../../constants";
import renderer from "react-test-renderer";

describe("Forgot Password screen", () => {
  const onSubmit = jest.fn();
  const forgotPassword = shallow(<ForgotPassword onSubmit={onSubmit} />);
  it("matches snapshot", () => {
    const screen = renderer.create(<ForgotPassword onSubmit={onSubmit} />);
    const tree = screen.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("returns a GradientView", () => {
    const gradient = forgotPassword.find("GradientView");
    expect(gradient).toHaveLength(1);
  });
  it("returns a KeyboardAvoidingView with container style", () => {
    const keyboardAvoid = forgotPassword.find("KeyboardAvoidingView");
    expect(keyboardAvoid).toHaveLength(1);
    expect(keyboardAvoid.prop("style")).toEqual(styles.container);
  });
  it("returns a Form with form styles", () => {
    const form = forgotPassword.find("Form");
    expect(form).toHaveLength(1);
    expect(form.prop("style")).toEqual(styles.form);
  });
  it("sets form initialValues to [{id: '1', value:''}, {id: '2', value: false}]", () => {
    const form = forgotPassword.find("Form");
    const initial = [{ id: "1", value: "" }, { id: "2", value: false }];
    expect(form.prop("initialValues")).toEqual(initial);
  });
  it("sets form onSubmit to onSubmit prop", () => {
    const form = forgotPassword.find("Form");
    expect(form.prop("onSubmit")).toEqual(onSubmit);
  });
  describe("isError prop", () => {
    it("renders an INVALID_EMAIL error on TextInput when true", () => {
      const onChange = jest.fn();
      const onSubmit = jest.fn();
      const forgotPassword = shallow(
        <ForgotPassword onSubmit={onSubmit} isError={true} />
      );
      const form = forgotPassword.find("Form");
      const formChildren = form
        .props()
        .children(
          [{ id: "1", value: "" }, { id: "2", value: false }],
          onChange,
          onSubmit
        );
      const textInput = formChildren.props.children[0];
      expect(textInput.props.errorText).toBe(INVALID_EMAIL);
    });
  });
  describe("Form children", () => {
    describe("TextInput", () => {
      it("returns a TextInput with label Email", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const emailInput = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[0];
        expect(emailInput.type.name).toBe("TextInput");
        expect(emailInput.props.label).toBe("Email");
      });
      it("has input styles", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const emailInput = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[0];
        expect(emailInput.props.style).toEqual(styles.input);
      });
      it("sets TextInput value equal to first value in values array", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const textInput = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "new value" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[0];
        expect(textInput.props.value).toBe("new value");
      });
      it("calls provided onChange function with 1 and new value onChange", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const textInput = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "new value" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[0];
        textInput.props.onChange("new value");
        expect(onChange).toHaveBeenCalledWith("1", "new value");
      });
    });
    describe("Button", () => {
      it("returns a Button with value Send Email", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const button = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[1];
        expect(button.type.name).toBe("Button");
        expect(button.props.value).toBe("Send Email");
      });
      it("has success and textColor white", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const button = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[1];
        expect(button.props.success).toBe(true);
        expect(button.props.textColor).toBe(white);
      });
      it("calls provided onSubmit onPress", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn(() => Promise.resolve());
        const button = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[1];
        button.props.onPress({ id: "2", value: "email@example.com" });
        expect(onSubmit).toHaveBeenCalled();
      });
    });
  });
  describe("Submitting", () => {
    describe("when loading", () => {
      it("replaces the button with an ActivityIndicator", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn();
        const activityIndicator = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: true }],
            onChange,
            onSubmit
          ).props.children[1];
        expect(activityIndicator.type.render.name).toBe("ActivityIndicator");
      });
    });
    describe("onSubmit process", () => {
      it("calls onChange with 2 and true", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn(() => Promise.resolve());
        const button = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[1];
        const email = "email@example.com";
        button.props.onPress({ id: "2", value: email });
        expect(onChange).toHaveBeenCalledWith("2", true);
      });
      it("calls onSubmit with the provided email", () => {
        const onChange = jest.fn();
        const onSubmit = jest.fn(() => Promise.resolve());
        const button = forgotPassword
          .find("Form")
          .props()
          .children(
            [{ id: "1", value: "" }, { id: "2", value: false }],
            onChange,
            onSubmit
          ).props.children[1];
        const email = "email@example.com";
        button.props.onPress({ id: "2", value: email });
        expect(onSubmit).toHaveBeenCalledWith(email);
      });
      describe("when onSubmit rejects", () => {
        it("calls onChange with 2 and false", () => {
          const onChange = jest.fn();
          const onSubmit = jest.fn(() => Promise.reject());
          const button = forgotPassword
            .find("Form")
            .props()
            .children(
              [{ id: "1", value: "" }, { id: "2", value: false }],
              onChange,
              onSubmit
            ).props.children[1];
          const email = "email@example.com";
          button.props.onPress({ id: "2", value: email });
          return Promise.resolve()
            .then()
            .then(() => {
              expect(onChange).toHaveBeenCalledWith("2", false);
            });
        });
        it("calls AlertIOS with a NETWORK_ERROR", () => {
          const onChange = jest.fn();
          const onSubmit = jest.fn(() => Promise.reject());
          const button = forgotPassword
            .find("Form")
            .props()
            .children(
              [{ id: "1", value: "" }, { id: "2", value: false }],
              onChange,
              onSubmit
            ).props.children[1];
          const email = "email@example.com";
          const spy = jest.spyOn(AlertIOS, "alert");
          button.props.onPress({ id: "2", value: email });
          return Promise.resolve()
            .then()
            .then(() => {
              expect(spy).toHaveBeenCalledWith("Error!", NETWORK_ERROR);
            });
        });
      });
      describe("when onSubmit resolves", () => {
        it("calls onChange with 2 and false", () => {
          const onChange = jest.fn();
          const onSubmit = jest.fn(() => Promise.resolve());
          const button = forgotPassword
            .find("Form")
            .props()
            .children(
              [{ id: "1", value: "" }, { id: "2", value: false }],
              onChange,
              onSubmit
            ).props.children[1];
          const email = "email@example.com";
          button.props.onPress({ id: "2", value: email });
          return Promise.resolve().then(() => {
            expect(onChange).toHaveBeenCalledWith("2", false);
          });
        });
      });
    });
  });
});
