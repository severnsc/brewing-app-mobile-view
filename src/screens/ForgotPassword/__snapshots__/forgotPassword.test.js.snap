// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password screen matches snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "justifyContent": "center",
      "margin": 5,
      "top": 100,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomColor": undefined,
          "borderBottomWidth": 1,
          "marginVertical": 10,
        }
      }
    >
      <Text
        accessibilityLabel="Email"
        style={
          Object {
            "color": undefined,
          }
        }
      >
        Email
      </Text>
      <TextInput
        autoFocus={false}
        keyboardType="default"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        testID="forgotPasswordInput"
        value=""
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#37c172",
          "borderColor": "#e9ecee",
          "borderWidth": 1,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID="forgotPasswordButton"
    >
      <Text
        accessibilityLabel="Send Email"
        style={
          Object {
            "color": "#fff",
            "fontSize": undefined,
          }
        }
      >
        Send Email
      </Text>
    </View>
    ); }}
  </View>
</View>
`;
